php-frontend/src/main/java/org/sonar/php/tree/impl/statement/ForEachStatementTreeImpl.java

```diff
@@ -0,0 +1,228 @@
+      List<StatementTree> statements, InternalSyntaxToken endForEachToken, InternalSyntaxToken eosToken
+  ) {
+    this(Kind.ALTERNATIVE_FOREACH_STATEMENT, header, colonToken, statements, endForEachToken, eosToken);
+  }
+
+  private ForEachStatementTreeImpl(
+      Kind kind,
+      ForEachStatementHeader header, @Nullable InternalSyntaxToken colonToken,
+      List<StatementTree> statements, @Nullable InternalSyntaxToken endForEachToken, @Nullable InternalSyntaxToken eosToken
+  ) {
+    this.header = header;
+    this.colonToken = colonToken;
+    this.statements = statements;
+    this.endforeachToken = endForEachToken;
+    this.eosToken = eosToken;
+
+    this.KIND = kind;
+  }
+
+  @Override
+  public SyntaxToken foreachToken() {
+    return header.foreachToken();
+  }
+
+  @Override
+  public SyntaxToken openParenthesisToken() {
+    return header.openParenthesisToken();
+  }
+
+  @Override
+  public ExpressionTree expression() {
+    return header.expression();
+  }
+
+  @Override
+  public SyntaxToken asToken() {
+    return header.asToken();
+  }
+
+  @Nullable
+  @Override
+  public ExpressionTree key() {
+    return header.key();
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken doubleArrowToken() {
+    return header.doubleArrowToken();
+  }
+
+  @Override
+  public ExpressionTree value() {
+    return header.value();
+  }
+
+  @Override
+  public SyntaxToken closeParenthesisToken() {
+    return header.closeParenthesisToken();
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken colonToken() {
+    return colonToken;
+  }
+
+  @Override
+  public List<StatementTree> statements() {
+    return statements;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken endforeachToken() {
+    return endforeachToken;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken eosToken() {
+    return eosToken;
+  }
+
+  @Override
+  public Kind getKind() {
+    return KIND;
+  }
+
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    return Iterators.concat(
+        Iterators.forArray(foreachToken(), openParenthesisToken(), expression(), asToken(), key(), doubleArrowToken(), value(), closeParenthesisToken(), colonToken),
+        statements.iterator(),
+        Iterators.forArray(endforeachToken, eosToken));
+  }
+
+  @Override
+  public void accept(TreeVisitor visitor) {
+    visitor.visitForEachStatement(this);
+  }
+
+  /**
+   * Utility class hidden from API (it's mainly created to avoid duplication in grammar)
+   */
+  public static class ForEachStatementHeader implements Tree {
+
+    private final InternalSyntaxToken foreachToken;
+    private final InternalSyntaxToken openParenthesisToken;
+    private final ExpressionTree expression;
+    private final InternalSyntaxToken asToken;
+    private final ExpressionTree key;
+    private final InternalSyntaxToken doubleArrowToken;
+    private final ExpressionTree value;
+    private final InternalSyntaxToken closeParenthesisToken;
+
+    public ForEachStatementHeader(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)